File: 2022/01/cpp/solution.cpp
#include <cstdio>
#include <fstream>
#include <iostream>
#include <ostream>
#include <string>
#include <vector>

int part_1(std::vector<std::string> &lines) {
  int max = 0;
  int current = 0;
  for (int i = 0; i < lines.size(); i++) {
    std::string line = lines[i];

    if (line == "") {
      if (max < current) {
        max = current;
      }
      current = 0;
    } else {
      int number = std::stoi(line);
      current += number;
    }
  }

  return max;
}

int part_2(std::vector<std::string> &lines) { return 0; }

int main() {
  std::vector<std::string> lines;
  std::ifstream file("../input.txt");

  std::string line;
  while (std::getline(file, line)) {
    lines.push_back(line);
  }

  std::cout << "Part 1: " << part_1(lines) << std::endl;
  std::cout << "Part 2: " << part_2(lines) << std::endl;

  return 0;
}

File: 2022/01/python/solution.py


def part_1(lines):
    max_value = 0
    current = 0

    for line in lines:
        try:
            current += int(line)
        except ValueError:
            if current > max_value:
                max_value = current
            current = 0
    if current > max_value:
        max_value = current
    return max_value


def part_2(lines):
    leaderboard = []
    current = 0

    for line in lines:
        try:
            current += int(line)
        except ValueError:
            leaderboard.append(current)
            leaderboard.sort()
            if len(leaderboard) > 3:
                leaderboard.pop(0)
            current = 0

    leaderboard.append(current)
    leaderboard.sort()
    if len(leaderboard) > 3:
        leaderboard.pop(0)

    return sum(leaderboard)


if __name__ == "__main__":
    with open('../input.txt') as file:
        lines = file.readlines()
    print(f'Part 1: {part_1(lines)}')
    print(f'Part 2: {part_2(lines)}')
